packages/gateway/src/manager.ts

```diff
@@ -54,6 +55,160 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
+            async message(shard, payload) {
+              if (payload.t === 'READY') {
+                await gateway.resharding.updateGuildsShardId(
+                  (payload.d as DiscordReady).guilds.map((g) => g.id),
+                  shardId,
+                )
+              }
+            },
+          },
+          requestIdentify: async () => {
+            await gateway.identify(shardId)
+          },
+          shardIsReady: async () => {
+            logger.debug(`[Shard] Shard #${shardId} is ready`)
+            await delay(gateway.spawnShardDelay)
+            logger.debug(`[Shard] Resolving shard identify request`)
+            gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)!.identifyRequests.shift()?.()
+          },
+        })
+
+        if (gateway.preferSnakeCase) {
+          shard.forwardToBot = async (payload) => {
+            options.events?.message?.(shard, payload)
+          }
+        }
+
+        gateway.resharding.shards.set(shardId, shard)
+
+        const bucket = gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)
+        if (!bucket) return
+
+        return await new Promise((resolve) => {
+          // Mark that we are making an identify request so another is not made.
+          bucket.identifyRequests.push(resolve)
+          logger.debug(`[Gateway] identifying shard #(${shardId}).`)
+          // This will trigger identify and when READY is received it will resolve the above request.
+          shard?.identify().then(async () => {
+            // Tell the manager that this shard is online
+            return await gateway.resharding.shardIsPending(shard)
+          })
+        })
+      },
+      async updateGuildsShardId(guildIds, shardId) {
+        logger.warn(`[Resharding] Updating the following guild ids shard to #${shardId}: ${guildIds.join(', ')}`)
+      },
+      async shardIsPending(shard) {
+        // Save this in pending at the moment, until all shards are online
+        gateway.resharding.pendingShards.set(shard.id, shard)
+        logger.warn(`[Resharding] Shard #${shard.id} is now pending`)
+        // Check if all shards are now online.
+        if (gateway.totalShards > gateway.resharding.pendingShards.size) return
+        logger.warn(`[Resharding] All shards are now online.`)
+
+        // New shards start processing events
+        for (const shard of gateway.resharding.pendingShards.values()) {
+          shard.events.message = options.events?.message
+        }
+
+        // Old shards stop processing events
+        for (const shard of gateway.shards.values()) {
+          const oldHandler = shard.events.message
+
+          shard.events.message = async function (_, message) {
+            // Member checks need to continue but others can stop
+            if (message.t !== 'GUILD_MEMBERS_CHUNK') return
+            // Process only the chunking events
+            oldHandler?.(shard, message)
+          }
```

Comment:
Just wondering if anyone can tell me why "member checks need to continue".

Reply:
Because the guild members chunk event is shard specific and will be continuing to come through the old shard only